tools/BaseRunner.py

```diff
@@ -157,7 +157,8 @@ def get_version(self):
             cmd = self.get_version_cmd()
 
             proc = subprocess.Popen(
-                cmd,
+                " ".join(cmd),  # Required to go via shell
+                shell=True,
```

Comment:
Why exactly is that needed?

Reply:
shell=True is required so that it can run a program that is not a binary, e.g. anything in perl/python with a #! such as Verilator.  Perhaps there's another way to do this but that seems the standard practice.
Once you have shell=True the join is required or the command will be broken, the python manual also discloses to pass a single space delimited line.

Reply:
Using `shell=True` is usually not recommended. We deal with lots of auto-generation that include third-party repositories, so I'd rather we didn't do this (see: https://docs.python.org/2/library/subprocess.html#frequently-used-arguments)

Could we override `get_version_cmd` in `Verilator.py` instead?

I think the following should work:
```
def get_version_cmd(self):
    return ["perl", self.executable, "--version"]
```

Reply:
That doesn't work because then "verilator" won't be resolved using path.  I tried pushing using $SHELL, which works locally for me, we'll see if that passes regressions.

Reply:
Right, let's not dwell on it too much then, thanks for the changes!